configure.ac

```diff
@@ -331,21 +348,41 @@ if test x"$hashes_enabled" = x || test x"$hashes_enabled" = x,; then
 fi
 AC_SUBST([hashes_enabled])
 
+
 # If the traditional DES hash is disabled, then the obsolete APIs are
-# implicitly disabled.
+# implicitly disabled, except for the case stubs are requested.
 case "$hashes_enabled" in
     *,descrypt,*) ;;
     *)
        if test x"$COMPAT_ABI" != xno; then
-         AC_MSG_WARN(
-           [--enable-hashes=${hashes_selected} forces --enable-obsolete-api=no])
+         if test x"$enable_obsolete_api_enosys" != x0; then
+           AC_MSG_WARN(
+             [--enable-hashes=${hashes_selected} does NOT include "descrypt".
+              The crypt{,_r{,a,n}} functions will not be POSIX compliant, nor
+              fully backwards compatible.])
```

Comment:
POSIX does not require `crypt*` to support descrypt-based passphrase hashing (see http://pubs.opengroup.org/onlinepubs/9699919799/functions/crypt.html -- an implementation that always returns ENOSYS when the salt argument is in the specified form is valid).  I don't think this warning is necessary; someone who turns off DES knows what they're doing.

Reply:
Fixed.

Shall `crypt*` return `ENOSYS` when trying to use `descrypt`, when stubs are built and `descrypt` is disabled?